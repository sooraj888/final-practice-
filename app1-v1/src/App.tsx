import React from "react";
import logo from "./logo.svg";
import "./App.css";
import { Box } from "@material-ui/core";

function App() {
  return <Box className="App"></Box>;
}

export default App;
