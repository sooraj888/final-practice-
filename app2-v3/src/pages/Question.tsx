import {
  Box,
  Button,
  FormControl,
  FormControlLabel,
  FormLabel,
  Radio,
  RadioGroup,
  TextField,
} from "@material-ui/core";
import React, { useState } from "react";
import { NavigateFunction, useNavigate, useParams } from "react-router-dom";
import QuestionIndiacator from "../components/QuestionIndiacator";
import questionData from "./../questionData.json";

const Question = (): JSX.Element => {
  const navigate: NavigateFunction = useNavigate();
  const [answerArray, setAnswerArray] = useState<string[]>(Array(3).fill(""));

  let id: number = Number(useParams().id);

  if (!id) {
    id = 1;
  }

  if (id > 3 && id < 1) {
    return <div>Invalid Question</div>;
  }

  return (
    <Box>
      <QuestionIndiacator id={id} answerArray={answerArray} />
      <Box
        sx={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "center",
          alignItems: "center",
        }}
      >
        <div>{questionData[id - 1].question}</div>

        {((): JSX.Element => {
          switch (id) {
            case 1:
              return (
                <FormControl>
                  <TextField
                    value={answerArray[0]}
                    onChange={(e: any) => {
                      setAnswerArray((prev) => {
                        const updateArr = [...prev];

                        updateArr[0] = e.target.value;
                        return updateArr;
                      });
                    }}
                  ></TextField>
                </FormControl>
              );
            case 2:
              const arr2 = questionData[1].option;
              return (
                <FormControl>
                  <RadioGroup
                    value={answerArray[1]}
                    onChange={(e: any) => {
                      setAnswerArray((prev) => {
                        const updateArr = [...prev];

                        updateArr[1] = e.target.value;
                        return updateArr;
                      });
                    }}
                  >
                    {arr2.map((item: any, key: number) => {
                      return (
                        <FormControlLabel
                          key={key}
                          control={<Radio />}
                          value={item}
                          label={item}
                        ></FormControlLabel>
                      );
                    })}
                  </RadioGroup>
                </FormControl>
              );
            case 3:
              const arr3 = questionData[2].option;
              return (
                <FormControl>
                  <RadioGroup
                    value={answerArray[2]}
                    onChange={(e: any) => {
                      setAnswerArray((prev) => {
                        const updateArr = [...prev];

                        updateArr[2] = e.target.value;
                        return updateArr;
                      });
                    }}
                  >
                    {arr3.map((item: any, key: number) => {
                      return (
                        <FormControlLabel
                          key={key}
                          control={<Radio />}
                          value={item}
                          label={item}
                        ></FormControlLabel>
                      );
                    })}
                  </RadioGroup>
                </FormControl>
              );
            default:
              return <></>;
          }
        })()}
      </Box>
      <Box sx={{ position: "absolute", left: "45%" }}>
        <Button
          onClick={(): void => {
            navigate("/result", { state: answerArray });
          }}
        >
          Submit
        </Button>
      </Box>
    </Box>
  );
};

export default Question;
