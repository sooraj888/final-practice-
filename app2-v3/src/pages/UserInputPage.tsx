import { Box, Button, TextField } from "@material-ui/core";
import React, { useState } from "react";
import { NavigateFunction, useNavigate } from "react-router-dom";

const UserInputPage = (): JSX.Element => {
  const [name, setName] = useState<string>("");
  const navigate: NavigateFunction = useNavigate();
  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",

        alignItems: "center",
      }}
    >
      <Box
        sx={{
          padding: "20px",
          display: "flex",
          flexDirection: "column",
          justifyContent: "space-around",
          alignItems: "center",
          width: "200px",
          height: "300px",
          border: "2px solid black",
          borderRadius: "20px",
          margin: "30px",
        }}
      >
        <TextField
          placeholder="enter name"
          required
          onChange={(e: any) => {
            setName(e.target.value);
          }}
          value={name}
          variant={"outlined"}
        ></TextField>
        <Button
          variant="contained"
          onClick={(): void => {
            navigate("/1");
          }}
        >
          Enter the Exam
        </Button>
      </Box>
    </Box>
  );
};

export default UserInputPage;
