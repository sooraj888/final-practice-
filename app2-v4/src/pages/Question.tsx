import {
  Box,
  Button,
  Checkbox,
  FormControl,
  FormControlLabel,
  Radio,
  RadioGroup,
  TextField,
} from "@material-ui/core";
import React, { useEffect, useMemo, useState } from "react";
import { NavigateFunction, useNavigate, useParams } from "react-router-dom";
import QuestionIndicator from "../components/QuestionIndicator";
import questionData from "./../questionData.json";
const Question = (): JSX.Element => {
  const navigate: NavigateFunction = useNavigate();
  let id: number = Number(useParams().id);
  if (!id) {
    id = 1;
  }

  type answerArrayType = Array<string | Array<any>>;
  const [answerArray, setAnswerArray] = useState<answerArrayType>(
    Array(4).fill([])
  );
  const [question4, setQuestion4] = useState<any[]>([]);
  const handleCheckBox = (value: number) => {
    setQuestion4((prev: any) => {
      let updatearr = [...prev];
      if (!updatearr.includes(value)) {
        updatearr = [value, ...updatearr];
      } else {
        updatearr = updatearr.filter((item: any) => {
          return item !== value;
        });
      }
      return updatearr;
    });
  };

  useMemo(() => {
    setAnswerArray((prev: any) => {
      const updateArr = [...prev];
      updateArr[3] = question4;
      return updateArr;
    });
  }, [question4]);

  useMemo(() => {
    console.log(answerArray);
  }, [answerArray]);

  if (id > 4 && id < 1) {
    return <h1>Question Not Found</h1>;
  }
  return (
    <Box>
      <QuestionIndicator answerArray={answerArray} id={id} />
      <Box>
        {<h5>{questionData?.[id - 1]?.question}</h5>}

        {(() => {
          switch (id) {
            case 1:
              return (
                <TextField
                  value={answerArray[0]}
                  onChange={(e: any) => {
                    setAnswerArray((prev: any) => {
                      const updateArr = [...prev];
                      updateArr[0] = e.target.value;
                      return updateArr;
                    });
                  }}
                ></TextField>
              );
            case 2:
              const arr2: string[] = questionData?.[1].option;
              return (
                <FormControl>
                  <RadioGroup
                    value={answerArray[1]}
                    onChange={(e: any) => {
                      setAnswerArray((prev: any) => {
                        const updateArr = [...prev];
                        updateArr[1] = e.target.value;
                        return updateArr;
                      });
                    }}
                  >
                    {arr2?.map((item: any) => {
                      return (
                        <FormControlLabel
                          label={item}
                          key={item}
                          value={item}
                          control={<Radio />}
                        ></FormControlLabel>
                      );
                    })}
                  </RadioGroup>
                </FormControl>
              );
            case 3:
              const arr3: string[] = questionData?.[2].option;
              return (
                <FormControl>
                  <RadioGroup
                    value={answerArray[2]}
                    onChange={(e: any) => {
                      setAnswerArray((prev: any) => {
                        const updateArr = [...prev];
                        updateArr[2] = e.target.value;
                        return updateArr;
                      });
                    }}
                  >
                    {arr3?.map((item: any) => {
                      return (
                        <FormControlLabel
                          label={item}
                          key={item}
                          value={item}
                          control={<Radio />}
                        ></FormControlLabel>
                      );
                    })}
                  </RadioGroup>
                </FormControl>
              );
            case 4:
              const arr4: string[] = questionData?.[3].option;
              return (
                <FormControl>
                  {" "}
                  {arr4?.map((item: any): JSX.Element => {
                    return (
                      <FormControlLabel
                        checked={question4.includes(item)}
                        onClick={() => handleCheckBox(item)}
                        label={item}
                        key={item}
                        value={item}
                        control={<Checkbox />}
                      ></FormControlLabel>
                    );
                  })}
                </FormControl>
              );
          }
        })()}
      </Box>

      <Box>
        <Button
          onClick={(): void => {
            navigate("/result", { state: answerArray });
          }}
        >
          Submit
        </Button>
      </Box>
    </Box>
  );
};

export default Question;
