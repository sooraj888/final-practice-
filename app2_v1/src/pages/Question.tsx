import {
  Box,
  Button,
  Checkbox,
  FormControl,
  FormControlLabel,
  Radio,
  RadioGroup,
  TextField,
} from "@material-ui/core";
import React, { useEffect, useState } from "react";
import { useNavigate, useParams } from "react-router-dom";
import QuestionIndiactor from "../components/QuestionIndiactor";
import questionData from "./../questionData.json";

const Question = () => {
  const navigate = useNavigate();
  let id: number = Number(useParams().id);

  if (!id) {
    id = 1;
  }

  const [answerArr, setAnswerArr] = useState<any>(Array(5).fill([]));
  const [question5Arr, setQuestion5Arr] = useState<any>([]);

  const handleOnCheckBoxClick = (item: any) => {
    // console.log(item);
    setAnswerArr((prev: any) => {
      const updateArr = [...prev];
      if (updateArr[3].includes(item)) {
        updateArr[3] = updateArr[3].filter((value: any) => {
          return value != item;
        });
      } else {
        updateArr[3] = [...updateArr[3], item];
      }
      return updateArr;
    });
  };

  const handleOnMatchTheFolloin = (item: any) => {
    // set
    console.log(item);
    setQuestion5Arr([...question5Arr, item]);
  };

  useEffect(() => {
    console.log(answerArr);
  }, [answerArr]);

  useEffect(() => {
    setAnswerArr((prev: any) => {
      const updateArr = [...prev];
      updateArr[4] = question5Arr;
      return updateArr;
    });
  }, [question5Arr]);

  if (id > 5 || id < 1) {
    return <>Question Not Found</>;
  }
  return (
    <Box>
      <QuestionIndiactor answerArr={answerArr} id={id} />
      <Box
        sx={{
          display: "flex",
          justifyContent: "center",
          flexDirection: "column",
          alignItems: "center",
        }}
      >
        {questionData?.[id - 1]?.question}
        <Box>
          {(() => {
            switch (id) {
              case 1:
                return (
                  <FormControl>
                    <TextField
                      value={answerArr[0]}
                      onChange={(e: any) => {
                        setAnswerArr((prev: any) => {
                          const updateArr = [...prev];
                          updateArr[0] = e.target.value;
                          return updateArr;
                        });
                      }}
                    ></TextField>
                  </FormControl>
                );
              case 2:
                const arr2 = questionData?.[1]?.option;
                return (
                  <FormControl>
                    <RadioGroup
                      value={answerArr[1]}
                      onChange={(e: any) => {
                        setAnswerArr((prev: any) => {
                          const updateArr = [...prev];
                          updateArr[1] = e.target.value;
                          return updateArr;
                        });
                      }}
                    >
                      {arr2.map((item) => {
                        return (
                          <FormControlLabel
                            key={item}
                            control={<Radio />}
                            value={item}
                            label={item}
                          ></FormControlLabel>
                        );
                      })}
                    </RadioGroup>
                  </FormControl>
                );
              case 3:
                const arr3 = questionData?.[2]?.option;
                return (
                  <FormControl>
                    <RadioGroup
                      value={answerArr[2]}
                      onChange={(e: any) => {
                        setAnswerArr((prev: any) => {
                          const updateArr = [...prev];
                          updateArr[2] = e.target.value;
                          return updateArr;
                        });
                      }}
                    >
                      {arr3.map((item) => {
                        return (
                          <FormControlLabel
                            key={item}
                            control={<Radio />}
                            value={item}
                            label={item}
                          ></FormControlLabel>
                        );
                      })}
                    </RadioGroup>
                  </FormControl>
                );
              case 4:
                const arr4 = questionData?.[3]?.option;
                return (
                  <FormControl>
                    {arr4.map((item) => {
                      return (
                        <FormControlLabel
                          checked={answerArr[3].includes(item) ? true : false}
                          onClick={() => {
                            handleOnCheckBoxClick(item);
                          }}
                          key={item}
                          control={<Checkbox />}
                          value={item}
                          label={item}
                        ></FormControlLabel>
                      );
                    })}
                  </FormControl>
                );
              case 5:
                const arr5 = questionData?.[4].option;
                return (
                  <Box>
                    {arr5.map((item, id) => {
                      return (
                        <div key={id}>
                          <span>{questionData[4].questionOption[id]}</span>
                          <span>{question5Arr[id]}</span>
                          <span>
                            <Button
                              disabled={question5Arr.includes(item)}
                              onClick={() => handleOnMatchTheFolloin(item)}
                            >
                              {item}
                            </Button>
                          </span>
                        </div>
                      );
                    })}
                  </Box>
                );
            }
          })()}
        </Box>
        <Button
          onClick={() => {
            navigate("/result", { state: answerArr });
          }}
        >
          Submit
        </Button>
      </Box>
    </Box>
  );
};

export default Question;
