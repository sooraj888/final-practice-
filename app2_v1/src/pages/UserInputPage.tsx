import { Box, Button, TextField } from "@material-ui/core";
import React from "react";
import { useNavigate } from "react-router-dom";

const UserInputPage = () => {
  const navigate = useNavigate();
  const handleOnSubmit = (e: any) => {
    e.preventDefault();
    navigate("/question/1");
  };
  return (
    <Box component={"form"} onSubmit={handleOnSubmit}>
      <TextField variant="outlined" placeholder="name" required></TextField>
      <Button type="submit">Submit</Button>
    </Box>
  );
};

export default UserInputPage;
