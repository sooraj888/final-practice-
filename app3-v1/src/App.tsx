import React, { useEffect, useState } from "react";
import logo from "./logo.svg";
import "./App.css";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import NavBar from "./components/NavBar";
import PostListPage from "./pages/PostListPage";
import RawJson from "./pages/RawJson";
import PageNotFound from "./pages/PageNotFound";
import axios from "axios";

export type postDataType = Array<Array<any>>;

function App(): JSX.Element {
  const [postData, setPostData] = useState<postDataType>([]);

  const [page, setPage] = useState<number>(1);

  let count: number = 0;

  let myInterval: NodeJS.Timer;

  const fetchData = async (): Promise<any> => {
    const responce: any = await axios.get(
      `http://hn.algolia.com/api/v1/search?query=bar&page=${count}`
    );
    console.log(responce);
    if (!responce?.data?.exhaustiveNbHits) {
      setPostData((prev: postDataType): postDataType => {
        let updatedArr: postDataType = [...prev];
        updatedArr = [...updatedArr, responce?.data?.hits];
        return updatedArr;
      });
      count += 1;
    } else {
      clearInterval(myInterval);
    }
  };

  useEffect((): void => {
    fetchData();
    myInterval = setInterval(fetchData, 3000);
  }, []);

  useEffect((): void => {
    console.log(postData);
  }, [postData]);

  return (
    <div className="App">
      <NavBar />
      <BrowserRouter>
        <Routes>
          <Route
            path="/"
            element={
              <PostListPage
                postData={postData}
                page={page}
                setPage={setPage}
              ></PostListPage>
            }
          />
          <Route path="/rawJson" element={<RawJson />}></Route>
          <Route path="*" element={<PageNotFound />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
