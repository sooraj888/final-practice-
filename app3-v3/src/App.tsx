import React, { useEffect, useState } from "react";
import logo from "./logo.svg";
import "./App.css";
import axios, { AxiosResponse } from "axios";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import PostList from "./pages/PostList";
import RawJson from "./pages/RawJson";
import PageNotFound from "./pages/PageNotFound";
import NavBar from "./Componets/NavBar";

function App(): JSX.Element {
  const [data, setData] = useState<any[]>([]);
  let count: number = 0;
  const [page, setPage] = useState<number>(1);
  let intervel: NodeJS.Timer;

  const [error, setError] = useState<boolean>(false);
  const url: string = "https://hn.algolia.com/api/v1/search?query=bar&page=";

  const fetchData = async (): Promise<any> => {
    try {
      setError(false);
      const responce: AxiosResponse<any, any> = await axios.get(url + count);
      if (!responce?.data?.exhaustiveNbHits) {
        setData((prev): any[] => [...prev, responce?.data?.hits]);
        count += 1;
      } else {
        clearInterval(intervel);
      }
    } catch (e) {
      setError(true);
    } finally {
    }
  };

  useEffect((): void => {
    fetchData();
    intervel = setInterval(fetchData, 10000);
  }, []);
  useEffect((): void => {
    console.log(data);
  }, [data]);

  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route
            path="/"
            element={
              <PostList
                setPage={setPage}
                data={data}
                page={page}
                error={error}
              />
            }
          />
          <Route path="/rawJson" element={<RawJson />} />
          <Route path="*" element={<PageNotFound />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;

//         ` http://hn.algolia.com/api/v1/search?query=bar&page=${count}
