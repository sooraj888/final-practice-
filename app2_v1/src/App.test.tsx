import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";
import UserInputPage from "./pages/UserInputPage";
import { BrowserRouter } from "react-router-dom";
import Question from "./pages/Question";

// test("renders learn react link", () => {
//   const { container } = render(
//     <BrowserRouter>
//       <UserInputPage />
//     </BrowserRouter>
//   );
//   fireEvent.change(screen.getByPlaceholderText("name"), {
//     target: { value: "sooraj" },
//   });

//   expect(container).toMatchSnapshot();
//   expect(screen.getByPlaceholderText("name")).toHaveValue("sooraj");
// });

// test("", () => {
//   render(
//     <BrowserRouter>
//       <Question />
//     </BrowserRouter>
//   );
//   screen.debug();
// });

test("asd", () => {
  const { container } = render(
    <BrowserRouter>
      <Question />
    </BrowserRouter>
  );
  screen.debug();
  expect(container).toMatchSnapshot();
  fireEvent.click(screen.getByRole("button"));
  // expect(screen.getByText(/1+2/)).toBeInTheDocument();
});
