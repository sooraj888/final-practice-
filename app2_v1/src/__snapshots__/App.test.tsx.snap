// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`asd 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <a
        href="/question/1"
      >
        <div
          class="MuiBox-root MuiBox-root-3"
        >
          1
        </div>
      </a>
      <a
        href="/question/2"
      >
        <div
          class="MuiBox-root MuiBox-root-4"
        >
          2
        </div>
      </a>
      <a
        href="/question/3"
      >
        <div
          class="MuiBox-root MuiBox-root-5"
        >
          3
        </div>
      </a>
      <a
        href="/question/4"
      >
        <div
          class="MuiBox-root MuiBox-root-6"
        >
          4
        </div>
      </a>
      <a
        href="/question/5"
      >
        <div
          class="MuiBox-root MuiBox-root-7"
        >
          5
        </div>
      </a>
    </div>
    <div
      class="MuiBox-root MuiBox-root-8"
    >
      1+1
      <div
        class="MuiBox-root MuiBox-root-9"
      >
        <div
          class="MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Submit
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
